modules/apps/segments/segments-api/src/main/java/com/liferay/segments/util/SegmentsExperienceUtil.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.segments.util;
+
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.segments.constants.SegmentsWebKeys;
+import com.liferay.segments.service.SegmentsExperienceLocalServiceUtil;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Eudaldo Alonso
+ */
+public class SegmentsExperienceUtil {
+
+	public static long getSegmentsExperienceId(
+		HttpServletRequest httpServletRequest) {
+
+		long[] segmentsExperienceIds = GetterUtil.getLongValues(
+			httpServletRequest.getAttribute(
+				SegmentsWebKeys.SEGMENTS_EXPERIENCE_IDS));
+
+		if (ArrayUtil.isNotEmpty(segmentsExperienceIds)) {
+			return segmentsExperienceIds[0];
+		}
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		return SegmentsExperienceLocalServiceUtil.
+			fetchDefaultSegmentsExperienceId(themeDisplay.getPlid());
```

Comment:
We usually follow 2 approaches:

1. Create a Manager class, that is instantiated using the SegmentsExperienceLocalService.

2. Make getSegmentsExperienceId receive the SegmentsExperienceLocalService.

We usually prefer the first option, so we can test it easily, and all the dependencies are explicit.

Thanks!!!

Reply:
Hey @cgoncas,

Do you have an example of the Manager class?

Thanks!!

Reply:
For example the [AnalyticsReportsDataProvider](https://github.com/liferay/liferay-portal/blob/f4753776823a66409290376241d024ef5f6d44c0/modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/data/provider/AnalyticsReportsDataProvider.java), follow this pattern.